### ISSUE TITLE
Top-level "Docs" breadcrumb disappeared in default theme

### ISSUE THREAD

**Issue by micahellison** [mkdocs/mkdocs#2827]
When we updated to mkdocs 1.3, our accessibility tests started failing on all pages. It turns out the top-level "Docs" breadcrumb has disappeared in the default theme.

Before:
![image](https://user-images.githubusercontent.com/4383304/164948753-0be4ff1c-ad59-4166-9149-488514c054bf.png)

After:
![image](https://user-images.githubusercontent.com/4383304/164948770-35bca92a-8dc2-4347-806f-96dbc2c43f74.png)

It looks like the [A tag inside wy-breadcrumbs was changed so that it no longer has any content](https://github.com/mkdocs/mkdocs/commit/fa52d3ae71251ae1c760cbe9b31caced816573d1#diff-cfc3ac6c9be30c04eecea2e3c295f14149b7ecdf159cdac641e6421673bd2847R3), which is invalid HTML for a link, hence why [our accessibility tests failed](https://github.com/jrnl-org/jrnl/runs/6142269783?check_suite_focus=true).

**Comment by oprypin** [comment]
The "Docs" word was changed to a "home" logo in the upstream "readthedocs" theme and then that was ported to MkDocs project.

I don't observe the complete absence of anything in that spot. Maybe somehow you didn't reload the CSS so you don't see the home logo.

You may be right that it is not nice to not have any text inside an `<a>` tag but that's how it is in the upstream theme (note that there's no MkDocs involved at all in the site that I'm comparing against!) so it's not clear that something should be changed.

Comparison:

https://user-images.githubusercontent.com/371383/164970045-02313e0a-2639-4ca4-8ee4-38f31b588ef7.mp4

**Comment by wren** [comment]
Hi @oprypin, and thanks for your response.

> I don't observe the complete absence of anything in that spot. Maybe somehow you didn't reload the CSS so you don't see the home logo.



I think there might be a slight miscommunication. I believe that @micahellison was saying that the problem is that an empty `a` element is invalid html, and is also bad for accessibility (it causes problems with screen readers).

So, even if the display bug is fixed to successfully show the new icon, this empty element will still fail accessibility tests (like those linked in the OP), and html linters (due to invalid html). And it's those failures that I believe are the real issue, not necessarily just the display bug (which I believe to be a symptom more than the root cause).


> The "Docs" word was changed to a "home" logo in the upstream "readthedocs" theme and then that was ported to MkDocs project.

> You may be right that it is not nice to not have any text inside an <a> tag but that's how it is in the upstream theme (note that there's no MkDocs involved at all in the site that I'm comparing against!) so it's not clear that something should be changed.

I'm really sorry, but I didn't realize that this theme was downstream from another project. Are you saying that we should file this bug with them instead? Sorry if this was the wrong place to file this!

**Comment by oprypin** [comment]
@wren  Sorry for the very late reply.

I do believe that the original report complains *both* about the absence of anything there (which I am not able to reproduce) *and* about the accessibility warning (which I understand is problematic, but it *is* coming from upstream).

And I'll also finally point to the upstream that I mention:
https://github.com/readthedocs/sphinx_rtd_theme

[The problematic part `></a>`](https://github.com/mkdocs/mkdocs/blob/fa52d3ae71251ae1c760cbe9b31caced816573d1/mkdocs/themes/readthedocs/breadcrumbs.html#L3):

```html
    <li><a href="{{ nav.homepage.url|url }}" class="icon icon-home" alt="{% trans %}Docs{% endtrans %}"></a> &raquo;</li>
```

[comes directly from porting this](https://github.com/readthedocs/sphinx_rtd_theme/blob/1.0.0/sphinx_rtd_theme/breadcrumbs.html#L25):

```html
      <li><a href="{{ pathto(master_doc) }}" class="icon icon-home"></a> &raquo;</li>
```

[(still there on *master*)](https://github.com/readthedocs/sphinx_rtd_theme/blob/master/sphinx_rtd_theme/breadcrumbs.html)

**Comment by oprypin** [comment]
Anyway, what would people suggest to change this to? With a direct and explained suggestion, I'm open to deviating from upstream. Also feel free to check if this has been reported upstream and do so.

**Comment by wren** [comment]
@oprypin Sorry for the slow reply! These last few weeks really got away from me.

> Anyway, what would people suggest to change this to? With a direct and explained suggestion, I'm open to deviating from upstream. Also feel free to check if this has been reported upstream and do so.

In order to fix the invalid html, the `a` element must have some content,. Any content will do, so you could put a `&nbsp;` and call it done. In order to be a little friendlier for screen readers, though, you want the content to be related to what users see normally. So, since it's a home icon, maybe the content can be "Home". 

Another option would be to move the value of the alt attribute (`{% trans %}Docs{% endtrans %}`) to be inside of the link instead (this is actually where it was before this change happened upstream).

What do you think? Do any of those options work for you?

**Comment by oprypin** [comment]
So again to sum up.
* For [the original report](https://github.com/mkdocs/mkdocs/issues/2827#issue-1213474072) - I can't reproduce it.
* For adding some text into the hyperlink to satisfy a linter, if I literally add the text into that, it changes the outcome.
  ![image](https://user-images.githubusercontent.com/371383/179619532-df6e985b-7722-4a27-8395-bd05bf299914.png)
 
  I don't have frontend skills to know the best way how to add the text without actually adding it.
  Pull requests are welcome

**Comment by micahellison** [comment]
Hi @oprypin, sorry for leaving this open so long without a response.

I've finally figured out the original issue that you couldn't reproduce -- it was indeed on our end, with some custom CSS on that `icon-home` class showing a white icon on a white background. Hence why it seemed to "disappear" for us. My apologies for mixing this up!

The accessibility issue from the empty A tag is still a problem, however. I see that the RTD team has fixed it on [their version](https://github.com/readthedocs/sphinx_rtd_theme/blob/master/sphinx_rtd_theme/breadcrumbs.html) by using `aria-label` instead of `alt` for the tag. I'll rename this issue to address that issue specifically and submit a PR for that single change.
